### ISSUE TITLE
Bad Gateway error

### ISSUE THREAD

**Issue by pihltd** [CancerDataAggregator/cda-python#10]
I'm working on problem 2 (Find data from all patients who have been treated with "Radiation Therapy, NOS" and have both genomic and proteomic data).  It looks like this is a variant of one of the examples in that all the data marked with this treatment are in GDC and none in PDC, so a From query is used.  When I do the following, I consistently get a HTTP 502 Bad Gateway error:

```

treatment = Q('ResearchSubject.DiagnosisTreatment.type = "Radiation Therapy, NOS"')
gdc = Q('ResearchSubject.Specimen.identifier.system = "GDC"')
pdc= Q('ResearchSubject.Specimen.identifier.system = "PDC"')

q = pdc.From(treatment.And(gdc))
r = q.run()

print(r)

```

The example From query in the jupyter notebook (the Ovarian Serous Cystadenocarcinoma example) runs fine

**Comment by pshapiro4broad** [comment]
I was able to reproduce this after correcting a typo in your example: `DiagnosisTreatment` should be `Diagnosis.Treatment`.

I think the 502 error is caused by a timeout configured in the GKE cluster. I'm not sure yet which service has this timeout set, but it looks like it's a 30 second limit. So until this is fixed, any queries that take longer than 30s will generate a 502 error.

I'll create a ticket in the appropriate repo to fix this.

A workaround is to use `limit` when calling `run`. By default `run` will retrieve `1000` records. Using a smaller number will let the query complete faster, and all records can still be retrieved by using pagination. I was able to run this code successfully:

```python
treatment = Q('ResearchSubject.Diagnosis.Treatment.type = "Radiation Therapy, NOS"')
gdc = Q('ResearchSubject.Specimen.identifier.system = "GDC"')
pdc= Q('ResearchSubject.Specimen.identifier.system = "PDC"')

q = pdc.From(treatment.And(gdc))
r = q.run(limit=100)

print(r)
```
